File: docs/branch/python_ide/leetcode/300.py
class Solution:
    def lengthOfLIS(self, nums: list[int]) -> int:
        dp=[]
        for me in nums:
            best=0
            for i in range(len(dp)):
                if dp[i]>best and nums[i]<me:
                    best=dp[i]
            dp.append(1+best)
        
        return max(dp)

Penter=Solution()
data=[0,1,0,3,2,3]
print(Penter.lengthOfLIS(data))

File: leetcode/300.cpp
//did not finish

#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<int> dp;
    map<int,int> sorted;//key=total; value=minimum number in that total
    int lengthOfLIS(vector<int>& nums) {
        for(int i =0;i<nums.size();++i){
            int now_beat_child_length=0;
            for(auto child:sorted){
                
                cout<<child.first<<"  "<<child.second;
                if(child.first>now_beat_child_length && child.second){
                    return 0;
                }
            }
            dp.push_back(1+now_beat_child_length);
        }
        for(auto it:dp){
            cout<<it;
        }
        int best=dp[0];
        for(auto it:dp){
            if(it>best){
                best=it;
            }
        }
        return best;
    }
};

Solution Penter=Solution();
int main(){
    vector<int> data={10,9,2,5,3,7,101,18};
    Penter.lengthOfLIS(data);
    return 0;
}